src/main/webapp/app/course/learning-paths/services/base-api-http.service.ts

```diff
@@ -9,6 +9,23 @@ export abstract class BaseApiHttpService {
 
     private readonly baseUrl = 'api';
 
+    /**
+     * Debounces a function call to prevent it from being called multiple times in a short period.
+     * @param callback The function to debounce.
+     * @param delay The delay in milliseconds to wait before calling the function.
+     */
+    public static debounce<T extends unknown[]>(callback: (...args: T) => void, delay: number) {
```

Comment:
Could you add the return type of this function?

```suggestion
    public static debounce<T extends unknown[]>(callback: (...args: T) => void, delay: number): (...args: T) => void {
```